### PR TITLE
Issue 7 tests

### PR DESCRIPTION
**Pull request by stu204146-kiel**
Add tests for Issue 7

### CODE REVIEW

File: src/test/FreeC/IR/InliningTests.hs [hunk 1]
```diff
@@ -0,0 +1,90 @@
+-- | This moduule contains tests for FreeC.IR.Inlining
```
Comment (just95) [inline src/test/FreeC/IR/InliningTests.hs]: Use Haddock markup if your comments contain code or identifiers and end sentences with a period.

```suggestion
-- | This module contains tests for "FreeC.IR.Inlining".
```

File: src/test/FreeC/IR/InliningTests.hs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                                                , inlineFuncDecls
+                                                )
+import           FreeC.Monad.Class.Testable     ( shouldSucceedWith )
+import           FreeC.Test.Expectations
+import           FreeC.Test.Environment
+import           FreeC.Test.Parser              ( parseTestFuncDecl
+                                                , parseTestExpr
+                                                )
+
+-- | Test group for inlineExpr and inlineFuncDecls
```
Comment (just95) [inline src/test/FreeC/IR/InliningTests.hs]:
```suggestion
-- | Test group for 'inlineExpr' and 'inlineFuncDecls'.
```

File: src/test/FreeC/IR/InliningTests.hs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      expected <- parseTestExpr "if True then 42 else 24"
+      return (res `shouldBeSimilarTo` expected)
+  context "inlineFuncDecls" $ do
+    it "inlines a recursive function correctly" $ shouldSucceedWith $ do
+      _   <- defineTestTypeCon "Peano" 0
+      _   <- defineTestCon "Zero" 0 "Peano"
+      _   <- defineTestCon "Succ" 1 "Peano -> Peano"
+      add <-
+        parseTestFuncDecl
+          "add (m :: Peano) (n :: Peano) :: Peano = case m of { Zero -> n ; Succ m' -> Succ (add' m n) }"
```
Comment (just95) [inline src/test/FreeC/IR/InliningTests.hs]: This line looks like it exceeds the 80 character line length limit. Brittany does not break up long string automatically. You have to insert `++` manually.

```suggestion
      add <-
        parseTestFuncDecl
        $  "add (m :: Peano) (n :: Peano) :: Peano = case m of {"
        ++ "   Zero -> n;"
        ++ "   Succ m' -> Succ (add' m n)"
        ++ "}"
```

Also check remaining lines in this file.

**Review by just95 — CHANGES_REQUESTED** [review]
The tests look good to me. The pipeline is currently failing because `defineTestTypeCon` has a new argument for the constructor names.

**Review by just95 — APPROVED** [review]

**Review by jo-hanna1997 — APPROVED** [review]

**Pull request merged**